components/SearchBar/CommandPalette.tsx

```diff
@@ -0,0 +1,190 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+
+import {
+    Combobox,
+    ComboboxInput,
+    ComboboxOption,
+    ComboboxOptions,
+    Dialog,
+    DialogPanel,
+    DialogBackdrop,
+} from '@headlessui/react';
+import { MagnifyingGlassIcon } from '@heroicons/react/20/solid';
+import { ExclamationCircleIcon } from '@heroicons/react/24/outline';
+import { useEffect, useState } from 'react';
+import { apiRequest } from '@/utils/api-request';
+import { GlobalSearchDocument } from '@/generated/discovery/graphql';
+import { debounce } from '@/utils/debounce';
+
+export function CommandPalette() {
+    const [query, setQuery] = useState('');
+    const [open, setOpen] = useState(false);
+    const [results, setResults] = useState<Product[]>([]);
+    const router = useRouter();
+
+    useEffect(() => {
+        if (typeof window === 'undefined') {
+            return;
+        }
+
+        const handleCommandK = (event: KeyboardEvent) => {
+            if (event.key === 'k' && (event.metaKey || event.ctrlKey)) {
+                event.preventDefault();
+                setOpen(true);
+            }
+        };
+
+        window.addEventListener('keydown', handleCommandK);
+
+        return () => {
+            window.removeEventListener('keydown', handleCommandK);
+        };
+    }, []);
+
+    useEffect(() => {
+        if (query === '') {
+            return;
+        }
+        const debouncedSearch = debounce(async (term: string) => {
+            const response = await apiRequest(GlobalSearchDocument, { term });
+            const results = response?.data?.search?.hits ?? [];
+
+            // @ts-expect-error
+            setResults(results);
+        }, 150);
+
+        debouncedSearch(query);
```

Comment:
The debounce it does not seem to work. I tested it and it seems it fires on every key press

Uploading Screen Recording 2025-05-22 at 8.58.13.mov…

Reply:
Thanks for spotting this 🙏